portfolio/src/main/webapp/script.js

```diff
@@ -39,7 +39,7 @@ window.addEventListener("load", myInit, true); function myInit(){
     alert("No ID Selected!");
     window.location.replace('/');
     return;
-  } else {
+  } else if (id !== null) {
```

Comment:
Good catch! But could you put this in an or statement with line 38? Otherwise users will have an ugly URL

Reply:
id being null occurs when there is no "id" param in the url (when you are already at '/') so the url doesn't have &id=null or anything like that. I actually tried what you suggested first (id === "" || id === null) but that resulted in '/' infinitely alerting and then redirecting to itself, alerting again, etc.

Reply:
Updated if structure based on Quinn's suggestion so that id !==null is checked before checking if id is empty.